### DISCUSSION TITLE
How to create Message Box?

### DISCUSSION THREAD

**Discussion by ewwink** [ObaraEmmanuel/Formation#30] [Q&A] [answered]
on button click I want to create  Message Box with "OK" or  "Yes or No" button

**Comment by ObaraEmmanuel** [comment, marked as answer]
There is no way to directly do this on the designer yet. However, you can easily do this on the code side. You will need to attach a command to the button like so:

![Screenshot 2024-03-13 194145](https://github.com/ObaraEmmanuel/Formation/assets/36537378/fa9e2d0f-8930-4e0a-85a0-29b6be465498)

Note the name `show_message`.

You can then call the dialog in your code

```python
from formation.loader import AppBuilder
from tkinter.messagebox import askyesno


class App(AppBuilder):

    def __init__(self):
        super().__init__(path="test.xml")
        self.connect_callbacks(self)

    def show_message(self):
        result = askyesno("Test", "This is a test message box")
        print(result)


if __name__ == "__main__":
    App().mainloop()
```

The name of the method `show_message` must match the name you provided in the command option.
You can read more on message boxes [here](https://runestone.academy/ns/books/published/thinkcspy/GUIandEventDrivenProgramming/02_standard_dialog_boxes.html)

**Comment by ewwink** [comment]
OK thanks again
